client/src/components/Dashboard.js

```diff
@@ -3,65 +3,129 @@ import React, { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import jwt_decode from 'jwt-decode';
 import axios from 'axios';
+import { Container, Button, Typography,
+         Box, TextField, Dialog,
+         DialogActions, DialogContent, DialogContentText,
+         DialogTitle
+} from '@mui/material';
 
 import { connect } from 'react-redux';
 import { loadUser } from '../actions/auth';
+import Bill from './Bill';
 
-function Dashboard({ loadUser }) {
-	const navigate = useNavigate();
-	const [bills, setBills] = useState();
+function Dashboard( {loadUser} ) {
+    const navigate = useNavigate();
+    const [bills, setBills] = useState(null);
+    const [hasBills, setHasBills] = useState(false);
+    const [error, setError] = useState();
+    const [open, setOpen] = useState(false);
+    const [newStoreName, setNewStoreName] = useState('');
+    const [newAmount, setNewAmount] = useState('');
+    
+    // Use useEffect hook to fetch the user's data once they log in.
+    useEffect(() => {
+        const token = localStorage.getItem('token');
+        
+        if(token) {
+            loadUser(token);
+            const {user} = jwt_decode(token);
+            if(!user) {
+                localStorage.removeItem('token');
+                navigate('/login');
+            }
+            
+            getBills(token, user.id);
+        } else  {
+            navigate('/login');
+        }
 
-	useEffect(() => {
-		const token = localStorage.getItem('token');
+        // Query the backend and database to get all the bills.
+        async function getBills(token, userid) {
+            const url = `http://localhost:8000/app/bill/?userid=${userid}`;
+            const headers = { 'x-auth-token': token };
+            const res = await axios.get(url, { headers });
+            const {bills, error} = res.data;
+            
+            if(error)   setError(error);
+            
+            if(bills.length === 0) setHasBills(false);
+            else                   setBills(bills);
+        };
+    });
+    // [loadUser, navigate]
```

Comment:
Took out dependency array.
The main purpose of this useEffect hood is:
   1: Query the backend when the user first logs in.
   2: Refresh the dashboard page when the user adds a new bill.
Adding a dependency array will stop it from refreshing. Create a different useEffect hook for [loadUser, navigate].